src/Layouts/Sidebars/Partials/LayerElement.vala

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2021 Alecaddd (https://alecaddd.com)
+ *
+ * This file is part of Akira.
+ *
+ * Akira is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * Akira is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with Akira. If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Authored by: Alessandro "Alecaddd" Castellani <castellani.ale@gmail.com>
+ */
+
+/*
+ * Layout component containing the layer UI widget, which adapts based on the
+ * node's model type (Artboard, Layer, Group).
+ */
+public class Akira.Layouts.Sidebars.Partials.LayerElement : Gtk.ListBoxRow {
+    public unowned Lib2.ViewCanvas view_canvas { get; construct; }
+    public unowned Lib2.Items.ModelInstance node { get; construct; }
+
+    private Gtk.Label label;
+
+    public LayerElement (Lib2.Items.ModelInstance instance, Lib2.ViewCanvas canvas) {
+        Object (
+            view_canvas: canvas,
+            node: instance
```

Comment:
call it instance instead of node? :P  Maybe cloning the instance may be better. But not sure if we need to keep the instance or simply the id and any other data that is need (name?)

Reply:
Layers will need to trigger the hover effect, as well as selecting the item, deleting it, changing the `name`, and updating the `lock` and `visibility` attributes.
What do you think would be better to dd to handle those things?